### PR TITLE
Liked watchlater

### PR DESCRIPTION
**Pull request by mohammadkhizerkhan**
added like feature
added watchlater feature
added dropdown for video card

### CODE REVIEW

File: src/components/SingleVideo.jsx [hunk 1]
```diff
@@ -1,29 +1,48 @@
...
+import React from "react";
+import { Link, useParams, useNavigate,useLocation } from "react-router-dom";
+import { useState, useEffect } from "react";
 import axios from "axios";
+import { useAuth, useLike, useWatchLater } from "../context";
+import { addToLike, removeFromLike } from "../services/LikedServices";
+import {
+  addToWatchLater,
+  removeFromWatchLater,
+} from "../services/WatchLaterServices";
```
Comment (znjs) [inline src/components/SingleVideo.jsx]: can add an index file for services also

File: src/components/SingleVideo.jsx [hunk 2]
```diff
@@ -100,20 +142,22 @@ function SingleVideo() {
           </button>
         </div>
         <div class="channel-div flex-row-center">
-            {/* <Link to={channelLink} target="_blank"> */}
-            <img src={channelImg} alt="" class="img-responsive video-channel-avatar"/>
-            {/* </Link> */}
-            <div class="channel-name font-2 font-bold">IGN</div>
+          {/* <Link to={channelLink} target="_blank"> */}
```
Comment (znjs) [inline src/components/SingleVideo.jsx:141]: can remove the commented code

File: src/reducer/index.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        return {
+          ...state,
+          liked:[...state.liked]
+        };
+    //   case ACTION_TYPE.CLEAR_CART:
+    //     console.log(action.payload)
+    //     return {
+    //       ...state,
+    //       cart: action.payload,
+    //     };
```
Comment (znjs) [inline src/reducer/index.js]: can remove the commented code

File: src/reducer/index.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        return {
+          ...state,
+          watchLater:[...state.watchLater]
+        };
+    //   case ACTION_TYPE.CLEAR_CART:
+    //     console.log(action.payload)
+    //     return {
+    //       ...state,
+    //       cart: action.payload,
+    //     };
```
Comment (znjs) [inline src/reducer/index.js]: can remove the commented code

File: src/services/LikedServices.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+//           },
+//         }
+//       );
+//       console.log(res)
+//       res.status===200 && cartDispatch({type:ACTION_TYPE.CLEAR_CART,payload:[]});
+//     } catch (error) {
+//         console.log("error in remove from cart",error)
+//     }
+//   })
+// }
```
Comment (znjs) [inline src/services/LikedServices.js]: can remove the commented code

File: src/components/Sidebar.jsx [hunk 6]
```diff
@@ -4,8 +4,8 @@ import { NavLink } from 'react-router-dom'
 function Sidebar() {
   const getActiveStyle = ({ isActive }) => {
     return {
-      color: isActive ? "#dade00" : "#dadada",
-      borderLeft:isActive? "4px solid #dade00":"",
+      color: isActive ? "#FFD300" : "#dadada",
```
Comment (muskaanshah) [inline src/components/Sidebar.jsx:7]: You can use color variables here

**Review by znjs — CHANGES_REQUESTED** [review]
- Good folder Structure
- Good naming convention used
- feature preview working as expected

**Review by ScoobyDubeyDo — APPROVED** [review]

File: src/components/SingleVideo.jsx [hunk 7]
```diff
@@ -36,30 +55,70 @@ function SingleVideo() {
...
+          {watchLaterState.watchLater.some(
+            (watchlater) => watchlater._id === video._id
+          ) ? (
+            <button
+              class="btn btn-s flex-row-center single-video-btn"
+              onClick={() =>
+                removeFromWatchLater(token, video, watchLaterDispatch)
+              }
+            >
+              <svg width="2rem" height="2rem" viewBox="0 0 24 24">
```
Comment (muskaanshah) [inline src/components/SingleVideo.jsx:95]: Suggestion: You can keep the common svgs in a separate file and import them if they are used multiple times

File: src/components/SingleVideo.jsx [hunk 8]
```diff
@@ -100,20 +142,22 @@ function SingleVideo() {
...
         </div>
         <div class="channel-div flex-row-center">
-            {/* <Link to={channelLink} target="_blank"> */}
-            <img src={channelImg} alt="" class="img-responsive video-channel-avatar"/>
-            {/* </Link> */}
-            <div class="channel-name font-2 font-bold">IGN</div>
+          {/* <Link to={channelLink} target="_blank"> */}
+          <img
+            src={channelImg}
+            alt=""
```
Comment (muskaanshah) [inline src/components/SingleVideo.jsx:144]: Don't leave alt blank

File: src/services/LikedServices.js [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    );
+    likeDispatch({type:ACTION_TYPE.REMOVE_FROM_LIKE,payload:video});
+  } catch (error) {
+      console.log("error in remove from like",error)
+  }
+
+};
+
+
+// const clearCart=(token,carts,cartDispatch)=>{
```
Comment (muskaanshah) [inline src/services/LikedServices.js]: Can remove commented code if not required

File: src/services/LikedServices.js [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+        `/api/user/likes/${video._id}`,
+      {
+        headers: {
+          authorization: token,
+        },
+      }
+    );
+    likeDispatch({type:ACTION_TYPE.REMOVE_FROM_LIKE,payload:video});
+  } catch (error) {
+      console.log("error in remove from like",error)
```
Comment (muskaanshah) [inline src/services/LikedServices.js:37]: can also use console.error if you want to display errors

**Review by muskaanshah — APPROVED** [review]

**Pull request merged**